client/metrics/metrics.go

```diff
@@ -0,0 +1,262 @@
+	CmdDurationTSO                      prometheus.Observer
+	CmdDurationTSOAsyncWait             prometheus.Observer
+	CmdDurationGetRegion                prometheus.Observer
+	CmdDurationGetAllMembers            prometheus.Observer
+	CmdDurationGetPrevRegion            prometheus.Observer
+	CmdDurationGetRegionByID            prometheus.Observer
+	CmdDurationScanRegions              prometheus.Observer
+	CmdDurationBatchScanRegions         prometheus.Observer
+	CmdDurationGetStore                 prometheus.Observer
+	CmdDurationGetAllStores             prometheus.Observer
+	CmdDurationUpdateGCSafePoint        prometheus.Observer
+	CmdDurationUpdateServiceGCSafePoint prometheus.Observer
+	CmdDurationScatterRegion            prometheus.Observer
+	CmdDurationScatterRegions           prometheus.Observer
+	CmdDurationGetOperator              prometheus.Observer
+	CmdDurationSplitRegions             prometheus.Observer
+	CmdDurationSplitAndScatterRegions   prometheus.Observer
+	CmdDurationLoadKeyspace             prometheus.Observer
+	CmdDurationUpdateKeyspaceState      prometheus.Observer
+	CmdDurationGetAllKeyspaces          prometheus.Observer
+	CmdDurationGet                      prometheus.Observer
+	CmdDurationPut                      prometheus.Observer
+	CmdDurationUpdateGCSafePointV2      prometheus.Observer
+	CmdDurationUpdateServiceSafePointV2 prometheus.Observer
+
+	CmdFailedDurationGetRegion                prometheus.Observer
+	CmdFailedDurationTSOWait                  prometheus.Observer
+	CmdFailedDurationTSO                      prometheus.Observer
+	CmdFailedDurationGetAllMembers            prometheus.Observer
+	CmdFailedDurationGetPrevRegion            prometheus.Observer
+	CmdFailedDurationGetRegionByID            prometheus.Observer
+	CmdFailedDurationScanRegions              prometheus.Observer
+	CmdFailedDurationBatchScanRegions         prometheus.Observer
+	CmdFailedDurationGetStore                 prometheus.Observer
+	CmdFailedDurationGetAllStores             prometheus.Observer
+	CmdFailedDurationUpdateGCSafePoint        prometheus.Observer
+	CmdFailedDurationUpdateServiceGCSafePoint prometheus.Observer
+	CmdFailedDurationLoadKeyspace             prometheus.Observer
+	CmdFailedDurationUpdateKeyspaceState      prometheus.Observer
+	CmdFailedDurationGet                      prometheus.Observer
+	CmdFailedDurationPut                      prometheus.Observer
+	CmdFailedDurationUpdateGCSafePointV2      prometheus.Observer
+	CmdFailedDurationUpdateServiceSafePointV2 prometheus.Observer
+
+	// RequestDurationTSO records the durations of the successful TSO requests.
+	RequestDurationTSO prometheus.Observer
+	// RequestFailedDurationTSO records the durations of the failed TSO requests.
+	RequestFailedDurationTSO prometheus.Observer
+)
+
+func initCmdDurations() {
+	// WithLabelValues is a heavy operation, define variable to avoid call it every time.
+	CmdDurationTSOWait = cmdDuration.WithLabelValues("wait")
```

Comment:
We can init them in L150-L199 directly

Reply:
It is by design that we have a function to do this so we can call `InitAndRegisterMetrics` to reinitialize them with a client option.